src/libs/ck-libs/ampi/ddt.C

```diff
@@ -12,7 +12,9 @@ CkDDT::pup(PUP::er &p) noexcept
   if (p.isPacking()) {
     types.resize(numTypes);
     for (int i=0; i<userTypeTable.size(); i++) {
-      types[i] = userTypeTable[i] == nullptr ? MPI_DATATYPE_NULL : std::max(AMPI_MAX_PREDEFINED_TYPE, std::min(CkDDT_MAX_PREDEFINED_TYPE, userTypeTable[i]->getType()));
+      types[i] = userTypeTable[i] == nullptr
+        ? MPI_DATATYPE_NULL
+        : std::max(AMPI_MAX_PREDEFINED_TYPE, std::min(CkDDT_FIRST_USER_TYPE, userTypeTable[i]->getType()));
```

Comment:
maybe do an if...then...else here

Reply:
Could you explain the logic here too? Why is it not just either MPI_DATATYPE_NULL or userTypeTable[i]->getType()?

Reply:
(maybe add the explanation as a comment)

Reply:
Reformatted and added a comment